packages/dataviews/src/dataviews-layouts/table/use-is-scrolled-end.ts

```diff
@@ -0,0 +1,62 @@
+/**
+ * WordPress dependencies
+ */
+import { useDebounce } from '@wordpress/compose';
+import { useCallback, useEffect, useRef, useState } from '@wordpress/element';
+import { isRTL } from '@wordpress/i18n';
+
+const isScrolledToEnd = ( element: Element ) => {
+	if ( isRTL() ) {
+		const scrollLeft = Math.abs( element.scrollLeft );
+		return scrollLeft <= 1;
+	}
+
+	return element.scrollLeft + element.clientWidth >= element.scrollWidth - 1;
+};
+
+export function useIsScrolledEnd( {
```

Comment:
I would love to have a doc block to understand what this hook does.

Reply:
Should it be named `useIsHorizontalScrollEnd` instead

Reply:
Oh yes, it would be better. Renamed by https://github.com/Automattic/wp-calypso/pull/103005/commits/e2c331fb0c7f15a4b809c9a664e95914deac59f8.